services/cd-service/pkg/repository/transformer.go

```diff
@@ -844,13 +844,18 @@ func (c *QueueApplicationVersion) Transform(ctx context.Context, state *State) (
 }
 
 type DeployApplicationVersion struct {
+	Authentication
 	Environment   string
 	Application   string
 	Version       uint64
 	LockBehaviour api.LockBehavior
 }
 
 func (c *DeployApplicationVersion) Transform(ctx context.Context, state *State) (string, error) {
+	err := state.checkUserPermissions(ctx, c.Environment, "*", auth.PermissionDeployRelease, c.RBACConfig)
```

Comment:
```suggestion
	err := state.checkUserPermissions(ctx, c.Environment, c.Application, auth.PermissionDeployRelease, c.RBACConfig)
```

Reply:
I think we need the app here. Not everyone is allowed to deploy all apps.

Reply:
Yes correct, I added the app check here.